### PR TITLE
`isotropic_fluorescent_thick_3d` model

### PR DESCRIPTION
**Pull request by talonchandler**
This PR is an `isotropic_fluorescent_thick_3d` model that uses torch and our new `calculate_transfer_function` and `apply_inverse_transfer_function` structure.

Highlights:
- if you would like to see the fluorescence optical transfer functions and a simple reconstruction, run `python examples/models/isotropic_fluorescent_thick_3d.py`
- there were a couple natural refactors along the way: a shared padding function that I moved to `utils`, and a shared forward model between `phase_thick_3d` and `isotropic_fluorescent_thick_3d`
- I added `NotImplementedErrors` to our TV reconstructions, since they haven't been moved to torch yet

@edyoshikun I particularly appreciate your review, since I know you're actively using fluorescence reconstructions.

### CODE REVIEW

**Review by edyoshikun — APPROVED** [review]
Tested this on napari for visual numerical inspection and LGTM. I think the next step would be to add it to recOrder once you make the appropriate changes to the interface on that side.  No rush on my side. It's not a blocker for me.

File: waveorder/util.py [hunk 1]
```diff
@@ -13,6 +13,64 @@
 
 numbers = re.compile(r"(\d+)")
 
+import torch
+
+
+def pad_zyx(zyx_data, z_padding):
```
Comment (ziw-liu) [inline waveorder/util.py:19]: Should this be `autp_pad_z` instead?

File: waveorder/models/isotropic_fluorescent_thick_3d.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        optical_transfer_function = optical_transfer_function[
+            z_padding:-z_padding
+        ]
+
+    # Very simple simulation, consider adding noise and bkg knobs
+    zyx_obj_hat = torch.fft.fftn(zyx_object)
+    zyx_data = zyx_obj_hat * optical_transfer_function
+    data = torch.real(torch.fft.ifftn(zyx_data))
+
+    data = torch.tensor(data + 10)  # Add a direct background
```
Comment (ziw-liu) [inline waveorder/models/isotropic_fluorescent_thick_3d.py:101]:
```suggestion
    data += 10  # Add a direct background
```

Isn't `data` already a tensor?

File: tests/test_util.py [hunk 3]
```diff
@@ -8,3 +9,38 @@ def test_gen_coordinate():
...
+
+
+def test_pad_zyx_negative_padding():
+    zyx_data = torch.zeros((3, 4, 5))
+    z_padding = -1
+    with pytest.raises(Exception):
+        util.pad_zyx(zyx_data, z_padding)
+
+
+def test_pad_zyx_zero_padding(zyx_data):
```
Comment (ziw-liu) [inline tests/test_util.py:27]:
```suggestion
def test_pad_zyx_no_padding(zyx_data):
```

File: waveorder/util.py (hunk 1 above)
Comment (talonchandler) [reply waveorder/util.py]: Thanks...poor name. Changed to `pad_zyx_along_z`.

File: waveorder/models/isotropic_fluorescent_thick_3d.py (hunk 2 above)
Comment (talonchandler) [reply waveorder/models/isotropic_fluorescent_thick_3d.py]: Thanks! Fixed in two places.

**Review by ziw-liu — APPROVED** [review]
Looks good, thanks!
